source/common/stats/allocator_impl.cc

```diff
@@ -316,5 +336,79 @@ Counter* AllocatorImpl::makeCounterInternal(StatName name, StatName tag_extracte
   return new CounterImpl(name, *this, tag_extracted_name, stat_name_tags);
 }
 
+void AllocatorImpl::forEachCounter(std::function<void(std::size_t)> f_size,
+                                   std::function<void(Stats::Counter&)> f_stat) const {
+  Thread::LockGuard lock(mutex_);
+  f_size(counters_.size());
+  for (auto& counter : counters_) {
+    f_stat(*counter);
+  }
+}
+
+void AllocatorImpl::forEachGauge(std::function<void(std::size_t)> f_size,
+                                 std::function<void(Stats::Gauge&)> f_stat) const {
+  Thread::LockGuard lock(mutex_);
+  size_t num_gauges = 0;
+  std::for_each(gauges_.begin(), gauges_.end(), [&num_gauges](Gauge* const gauge) mutable {
+    if (gauge->importMode() != Gauge::ImportMode::Uninitialized) {
+      num_gauges++;
+    }
+  });
+  f_size(num_gauges);
```

Comment:
sorry a few more nits. Can you skip this loop if f_size is nullptr?

Reply:
Done